library/alloc/src/slice.rs

```diff
@@ -199,6 +209,34 @@ mod hack {
             }
             vec
         }
+
+        #[inline]
+        default fn to_boxed_slice<A: Allocator>(s: &[Self], alloc: A) -> Box<[Self], A> {
+            struct DropGuard<T> {
```

Comment:
This is a lot of duplicated code. Could we implement the `to_vec` above in terms of `to_boxed_slice`?